_code/cell-typing-advanced-strategies/getSubtypingGenes.R

```diff
@@ -0,0 +1,165 @@
+message("Key functions:
+      - findSafeGenes: identify genes safe from excessive bias from segmentation errors.
+      - getSubclusteringGenes: identify highly variable genes in a counts matrix, for use in clustering.")
+
+
+#' Identify genes useful for unsupervised clustering of closely-related cell types 
+#' Intended for subclustering of a broad cell type, e.g. of immune cells into T-cells, B-cells, etc... 
+#' @param mat Expression matrix from just the cell type in question, cells in rows and genes in columns.
+#' @param xy Spatial coordinates of cells 
+#' @param varratiothresh Genes with observed variance / predicted variance above this threshold will be called HVGs. Higher = a more stringent filter.
+#' @param expressionthresh Only keep genes with average raw counts above this level in your cell type of interest. 
+#' @param loess.span Parameter guiding loess fit used in hvg search.
+#' @return A vector of gene named for use in unsupervised subclustering
+getSubclusteringGenes <- function(mat, varratiothresh = 1, expressionthresh = 0.2, loess.span = 0.3) {
+  
+  ## identify hvgs in the counts matrix:
+  RawTargetVar <- colvars(mat)^2
```

Comment:
it looks liek colvars is calculating `var`, so the result here doesnt need to be squared.

`RawTargetVar <- colvars(mat)^2`